### PR TITLE
update build specs

### PR DESCRIPTION
**Pull request by jcd496**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +45,15 @@ endif()
...
 
+
+find_package(OpenMP REQUIRED)
+foreach(lang IN ITEMS CXX C)
+  get_property(tmp TARGET OpenMP::OpenMP_${lang} PROPERTY INTERFACE_COMPILE_OPTIONS)
+  list(TRANSFORM tmp REPLACE "-fexceptions" "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcompiler >-fexceptions")
+  set_property(TARGET OpenMP::OpenMP_${lang} PROPERTY INTERFACE_COMPILE_OPTIONS "${tmp}")
+  unset(tmp)
+endforeach()
+
```
Comment (germasch) [inline CMakeLists.txt:56]: I'd rather not have OpenMP in there unless we really want to use it. Do you think you'll need it any time soon? Generally, the more dependencies, the more ways for things to go wrong...

File: spack/psc/packages/psc/package.py [hunk 2]
```diff
@@ -42,5 +42,7 @@ def cmake_args(self):
             'ON' if '+nvtx' in self.spec else 'OFF')]
         args += ['-DPSC_USE_RMM={}'.format(
             'ON' if '+rmm' in self.spec else 'OFF')]
+        args += ['-DBUILD_TESTING={}'.format(
+            'ON' if '+tests' in self.spec else 'OFF')]
```
Comment (germasch) [inline spack/psc/packages/psc/package.py:46]: This is good, but it also needs a corresponding `variant` to actually work.

File: src/libpsc/CMakeLists.txt [hunk 3]
```diff
@@ -65,6 +65,10 @@ endif()
 if (PSC_HAVE_RMM)
   target_link_libraries(psc PUBLIC rapids::rmm)
 endif()
+if (PSC_HAVE_NVTX)
+  target_link_libraries(mrc PUBLIC CUDA::nvToolsExt)
+  set(HAVE_NVTX 1)
+endif()
```
Comment (germasch) [inline src/libpsc/CMakeLists.txt:71]: Why is this needed? nvtx should already be pulled in via libmrc.

File: CMakeLists.txt (hunk 1 above)
Comment (jcd496) [reply CMakeLists.txt:56]: I put it in because there is one section of libpsc wrapped in parallel for that we can use.

File: spack/psc/packages/psc/package.py (hunk 2 above)
Comment (jcd496) [reply spack/psc/packages/psc/package.py:46]: Oh ya, I keep losing changes I've made bouncing around branches.  I forgot to add that when I put it in this pr branch

File: src/libpsc/CMakeLists.txt (hunk 3 above)
Comment (jcd496) [reply src/libpsc/CMakeLists.txt:71]: IDK previously when you first merged nvtx it wasn't included in libpsc and so I couldn't build until I added that.  I've been doing it ever since for new branches, maybe its not required now.

File: CMakeLists.txt (hunk 1 above)
Comment (germasch) [reply CMakeLists.txt:56]: So I think this part shouldn't be there standalone then, but in a PR that actually adds something that uses OpenMP.

Also, I'd guess that you copied the subsequent `foreach` stuff from MPI just because it was there, or did you in fact get errors due `-fexceptions` from nvcc?

File: src/libpsc/CMakeLists.txt (hunk 3 above)
Comment (germasch) [reply src/libpsc/CMakeLists.txt:71]: I think it's not, so please remove it unless you find that it is needed after all.

Sorry for being a PITA about these bits ;)

File: CMakeLists.txt (hunk 1 above)
Comment (jcd496) [reply CMakeLists.txt:56]: Oh no, I straight copied it.  maybe this would be better 
`find_package(OpenMP)
if(OpenMP_CXX_FOUND)
    target_link_libraries(MyTarget PUBLIC OpenMP::OpenMP_CXX)
endif()`
But either way we can just leave it out until there is openmp

File: src/libpsc/CMakeLists.txt (hunk 3 above)
Comment (jcd496) [reply src/libpsc/CMakeLists.txt:71]: haha np, it is your code after all

File: CMakeLists.txt (hunk 1 above)
Comment (germasch) [reply CMakeLists.txt]: Yes, that's the right way (with the link part moved to `src/libpsc`)
